File: src/main/java/com/codecool/cashtrack/domain/entities/Friend.java
package com.codecool.cashtrack.domain.entities;

import com.codecool.cashtrack.domain.entities.ids.FriendId;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name = "friends")
public class Friend {

    @EmbeddedId
    private FriendId id;

    @MapsId("user1")
    @ManyToOne
    @JoinColumn(name = "user_id_1", nullable = false)
    private User user1;

    @MapsId("user2")
    @ManyToOne
    @JoinColumn(name = "user_id_2", nullable = false)
    private User user2;

    public Friend(User a, User b) {
        if (a.getId() < b.getId()) {
            this.user1 = a;
            this.user2 = b;
        } else {
            this.user1 = b;
            this.user2 = a;
        }
        this.id = new FriendId(this.user1.getId(), this.user2.getId());
    }

    public Friend() {

    }
}

File: src/main/java/com/codecool/repvault/domain/entities/Friend.kt
package com.codecool.repvault.domain.entities

import com.codecool.repvault.domain.entities.ids.FriendId
import jakarta.persistence.*

@Entity
@Table(name = "friends")
class Friend {
    @EmbeddedId
    var id: FriendId? = null

    @MapsId("user1")
    @ManyToOne
    @JoinColumn(name = "user_id_1", nullable = false)
    var user1: User? = null

    @MapsId("user2")
    @ManyToOne
    @JoinColumn(name = "user_id_2", nullable = false)
    var user2: User? = null

    constructor(a: User, b: User) {
        if (a.id!! < b.id!!) {
            this.user1 = a
            this.user2 = b
        } else {
            this.user1 = b
            this.user2 = a
        }
        this.id = FriendId(this.user1!!.id!!, this.user2!!.id!!)
    }

    constructor()
}